spark-doris-connector/spark-doris-connector-it/src/test/java/org/apache/doris/spark/sql/DorisAnySchemaITCase.scala

```diff
@@ -0,0 +1,235 @@
+      (1, 0, "user5", 100),
+      (3, 0, "user6", 200)
+    )).toDF("siteid", "citycode", "username", "pv")
+    df.write
+      .format("doris")
+      .option("doris.fenodes", getFenodes)
+      .option("doris.table.identifier", DATABASE + "." + dorisTable)
+      .option("user", getDorisUsername)
+      .option("password", getDorisPassword)
+      .option("doris.sink.properties.format", "json")
+      .option("doris.sink.properties.partial_columns", "true")
+      .option("doris.write.fields", "siteid,citycode,username")
+      .option("sink.batch.size", 2)
+      .option("sink.max-retries", 2)
+      .save()
+    spark.stop()
+  }
+
+  @Test
+  def jsonDataWriteSqlTest(): Unit = {
+    initializeTable(dorisTable, DataModel.UNIQUE)
+    initializeTable(dorisSourceTable, DataModel.UNIQUE)
+    val spark = SparkSession.builder().master("local[*]").getOrCreate()
+    val doris = spark.sql(
+      s"""
+         |CREATE TEMPORARY VIEW test_lh
+         |USING doris
+         |OPTIONS(
+         | "table.identifier"="${DATABASE + "." + dorisTable}",
+         | "fenodes"="${getFenodes}",
+         | "user"="${getDorisUsername}",
+         | "password"="${getDorisPassword}"
+         |)
+         |""".stripMargin)
+    spark.sql(
+      """
+        |insert into test_lh values (0, 0, "user1", 100), (1, 0, "user2", 100),(2, 1, "user3", 100),(3, 0, "user2", 200)
+        |""".stripMargin)
+
+    spark.sql(
+      s"""
+         |CREATE TEMPORARY VIEW table2
+         |USING doris
+         |OPTIONS(
+         | "table.identifier"="${DATABASE + "." + dorisSourceTable}",
+         | "fenodes"="${getFenodes}",
+         | "user"="${getDorisUsername}",
+         | "password"="${getDorisPassword}"
+         |)
+         |""".stripMargin)
+
+    spark.sql(
+      """
+        |insert into table2 values (0, 0, "user1", 100), (1, 0, "user2", 100),(2, 1, "user3", 100),(3, 0, "user2", 200)
+        |""".stripMargin)
+
+    spark.sql(
+      """
+        |insert into test_lh select siteid, citycode + 1, username, pv + 1 from table2
+        |""".stripMargin)
+
+    spark.stop()
+  }
+
+  @Test
+  def jsonDataWriteWithPartialUpdateSqlTest(): Unit = {
+    initializeTable(dorisTable, DataModel.UNIQUE)
+    val spark = SparkSession.builder().master("local[*]").getOrCreate()
+    val doris = spark.sql(
+      s"""
+         |CREATE TEMPORARY VIEW test_lh
+         |USING doris
+         |OPTIONS(
+         | "table.identifier"="${DATABASE + "." + dorisTable}",
+         | "fenodes"="${getFenodes}",
+         | "user"="${getDorisUsername}",
+         | "password"="${getDorisPassword}",
+         | "doris.sink.properties.format" = "csv",
+         | "doris.sink.properties.partial_columns" = "true",
+         | "doris.write.fields" = "siteid,citycode,username"
+         |)
+         |""".stripMargin)
+
+    spark.sql(
+      """
+        |insert into test_lh(siteid,citycode,username) values (0, 0, "user1"), (1, 0, "user2")
+        |""".stripMargin)
+    spark.stop()
```

Comment:
It is best to put it in finally. Otherwise, after the case fails to execute, `spark.stop` will not be executed, which may cause errors later.